### PR TITLE
[plugins/range] Various fixes

### PR DESCRIPTION
**Pull request by insomniacslk**
* removed unused variables
* removed some code duplication
* various errors/logging fixes

Signed-off-by: Andrea Barberio <insomniac@slackware.it>

### CODE REVIEW

**Comment by insomniacslk** [comment]
CC @borna-blazevic

**Comment by insomniacslk** [comment]
usual thing for `tip`

File: plugins/range/plugin.go [hunk 1]
```diff
@@ -31,34 +32,26 @@ type Record struct {
...
-}
-
-// LoadDHCPv4Records loads the DHCPv4Records global map with records stored on
+// loadRecords loads the DHCPv6/v4 Records global map with records stored on
 // the specified file. The records have to be one per line, a mac address and an
-// IPv4 address.
-func LoadDHCPv4Records(filename string) (map[string]*Record, error) {
-	log.Printf("reading leases from %s", filename)
+// IP address.
+func loadRecords(r io.Reader, v6 bool) (map[string]*Record, error) {
```
Comment (pmazzini) [inline plugins/range/plugin.go]: The 1st parameter (io.Reader) is not being used.

File: plugins/range/plugin.go [hunk 2]
```diff
@@ -133,41 +132,52 @@ func setupRange4(args ...string) (handler.Handler4, error) {
...
 	if binary.BigEndian.Uint32(ipRangeStart.To4()) >= binary.BigEndian.Uint32(ipRangeEnd.To4()) {
-		return nil, nil, errors.New("start of IP range has to be lower than the end fo an IP range")
+		return nil, nil, errors.New("start of IP range has to be lower than the end of an IP range")
 	}
 	LeaseTime, err = time.ParseDuration(args[3])
 	if err != nil {
-		return Handler6, Handler4, errors.New("expected an uint32, got: " + args[3])
+		return Handler6, Handler4, fmt.Errorf("invalid duration: %v", args[3])
+	}
+	r, err := os.Open(filename)
```
Comment (pmazzini) [inline plugins/range/plugin.go]: This file is not being closed and I think it is not being used either.

File: plugins/range/plugin.go (hunk 1 above)
Comment (insomniacslk) [reply plugins/range/plugin.go]: fixed

File: plugins/range/plugin.go (hunk 2 above)
Comment (insomniacslk) [reply plugins/range/plugin.go]: related to the above

File: plugins/range/plugin.go (hunk 1 above)
Comment (insomniacslk) [reply plugins/range/plugin.go]: another indicator that using global variables is rarely a good idea...

File: plugins/range/plugin.go [hunk 3]
```diff
@@ -31,41 +31,25 @@ type Record struct {
...
-	if err != nil {
-		return nil, err
-	}
+// IP address.
+func loadRecords(r io.Reader, v6 bool) (map[string]*Record, error) {
+	sc := bufio.NewScanner(r)
 	records := make(map[string]*Record)
-	for _, lineBytes := range bytes.Split(data, []byte{'\n'}) {
-		line := string(lineBytes)
+	for sc.Scan() {
```
Comment (pmazzini) [inline plugins/range/plugin.go]: Should the same thing be done in the `file` plugin for consistency? https://github.com/coredhcp/coredhcp/blob/master/plugins/file/plugin.go#L78

Comment (insomniacslk) [reply plugins/range/plugin.go]: yes, but in another PR, I don't want to stash too many changes here.

My plan is to move the commob stuff to `pkg`, and rebase `file` and  `range` on it

**Pull request merged**